SearchInRotatedSortedArray.java

```diff
@@ -0,0 +1,40 @@
+//time - o(log n) with constant space
+//please help me with nums[] having duplicates, I am thinking that <= and >= will handle it but it is not
```

Comment:
https://github.com/devashishnyati/Binary-Search-1/blob/master/binarysearchinrotatedarray.py
Look at this code. Let me know if you need any help.